src/phoenix/schemas/core.py

```diff
@@ -0,0 +1,62 @@
+from dataclasses import dataclass
+from datetime import datetime
+from typing import Any, Dict, Optional
+
+from phoenix.db import models
+from sqlalchemy import case, func, select
+from sqlalchemy.ext.asyncio import AsyncSession
+
+
+@dataclass
+class Dataset:
+    id: int
+    name: str
+    description: Optional[str]
+    created_at: datetime
+    updated_at: datetime
+    metadata: Dict[str, Any]
+
+    @classmethod
+    def from_model(cls, model: models.Dataset) -> "Dataset":
+        return cls(
+            id=model.id,
+            name=model.name,
+            description=model.description,
+            created_at=model.created_at,
+            updated_at=model.updated_at,
+            metadata=model.metadata_,
+        )
+
+    async def count_active_examples(self, session: AsyncSession) -> int:
```

Comment:
Feels like we're kind of hiding the query that's being made so it's not easily visible from the route.

Reply:
just as a matter of preference I do like the route handling function to only deal with the control flow for what kind of response to return instead of also including model layer logic, but this is being refactored anyway